scripts/ostrta.coffee

```diff
@@ -0,0 +1,85 @@
+# Description:
+#   One Script To Rule Them All
+#
+# Dependencies:
+#   None
+#
+# Configuration:
+#   None
+#
+# Commands:
+#   hubot listenFor <key> <percent%?> <response>
+#   hubot respondTo <key> <percent%?> <response>
+#
+# Notes:
+#
+# Author:
+#   maschall
+
+_ = require 'underscore'
+
+commandMap =
+  listenFor: "hear"
+  respondTo: "respond"
+
+brainKey = (command, key = "", topic = "") ->
+  "ostrta-#{command}#{topic}-#{key.replace(/\s/g, '_')}"
+
+add = (robot, key, value) ->
+  objects = fetch(robot, key)
+  if !objects
+    objects = []
+  objects.push value
+  remember(robot, key, objects)
+
+remember = (robot, key, value) ->
+  robot.brain.set(key, value)
+  
+fetch = (robot, key) ->
+  robot.brain.get(key)
+  
+percentFromString = (percentString) ->
+  parseInt(percentString, 10)
+  
+tellRobotWhatToDo = (robot, command, key) ->
+  robot[commandMap[command]](new RegExp("(#{key}(\\s|$))", 'i'), (msg) ->
```

Comment:
so I was thinking if you could close of the " in the regex early than you probably could inject something, however the input regex would capture the ".  So I tried sending in the ascii char and a unicode char, but hipchat sends those as just plain characters.  I don't think you can close off the " in the regexp constructor, but I could also be dead wrong.